components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -2509,34 +2509,45 @@ public static void addDefinedAllSequencesToRegistry(UserRegistry registry,
 	                                                    String customSequenceType)
 			throws APIManagementException {
 
+		InputStream inSeqStream = null;
+		String seqFolderLocation =
+				APIConstants.API_CUSTOM_SEQUENCES_FOLDER_LOCATION + File.separator +
+				customSequenceType;
+
 		try {
-			File inSequenceDir = new File(
-					APIConstants.API_CUSTOM_SEQUENCES_FOLDER_LOCATION + File.separator +
-					customSequenceType);
-			File[] sequences = inSequenceDir.listFiles();
-
-			for (File sequenceFile : sequences) {
-				String sequenceFileName = sequenceFile.getName();
-				String regResourcePath =
-						APIConstants.API_CUSTOM_SEQUENCE_LOCATION + File.separator +
-						customSequenceType + File.separator + sequenceFileName;
-				if (registry.resourceExists(regResourcePath)) {
-					if (log.isDebugEnabled()) {
-						log.debug("Defined sequences have already been added to the registry");
-					}
-				} else {
-					if (log.isDebugEnabled()) {
-						log.debug("Adding defined sequences to the registry.");
+			File inSequenceDir = new File(seqFolderLocation);
+			File[] sequences;
+			sequences = inSequenceDir.listFiles();
+
+			if (sequences != null) {
+				for (File sequenceFile : sequences) {
+					String sequenceFileName = sequenceFile.getName();
+					String regResourcePath =
+							APIConstants.API_CUSTOM_SEQUENCE_LOCATION + File.separator +
+							customSequenceType + File.separator + sequenceFileName;
+					if (registry.resourceExists(regResourcePath)) {
+						if (log.isDebugEnabled()) {
+							log.debug("Defined sequences have already been added to the registry");
```

Comment:
Please note this is already fixed in #1534 (https://github.com/wso2/carbon-apimgt/commit/d4afe424504cc45452116e2669842c8639f38f98)